sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVInferSchema.scala

```diff
@@ -160,6 +169,15 @@ class CSVInferSchema(val options: CSVOptions) extends Serializable {
   private def tryParseDouble(field: String): DataType = {
     if ((allCatch opt field.toDouble).isDefined || isInfOrNan(field)) {
       DoubleType
+    } else {
+      tryParseTimestampNTZ(field)
```

Comment:
why try parsing as TimestampNTZ here?

Reply:
Oh, there is a parsing chain in CSV. It used to be "DoubleType -> TimestampType" and now it is "DoubleType -> TimestampNTZType -> TimestampType". This is the order in which we infer columns. So if we cannot parse the timestamp as TIMESTAMP_NTZ, we will try parsing it as a timestamp with timezone. Does it answer your question?

Reply:
Ah, I see

Reply:
When I supported timestamp inference in JSON https://github.com/apache/spark/pull/23201, I had to introduce new option in https://github.com/apache/spark/pull/23455, and disable the inference by default in https://github.com/apache/spark/pull/28966 because such inference slowed down user queries. If you are sure that  is not the case, please, extend CSV benchmarks to proof the performance doesn't degrade.